js/src/elements/form-behavior.ts

```diff
@@ -168,23 +170,25 @@ class FormBehaviorElement extends HTMLElement {
 		}
 
 		const form = event.currentTarget as HTMLFormElement;
-		const validationErrorEvent = new CustomEvent('validationerror', {
-			bubbles: true,
-			cancelable: true
-		});
 
 		// reject the submit if no edits have been made (overridable with the new attribute)
 		if (!this.canSave) {
 			this.showNoChangesMessage(form);
-			this.dispatchEvent(validationErrorEvent);
 			return;
 		}
 
 		try {
 			this.submitting = true;
 			setBusySubmitButton(form, this.submitting);
-			const valid = await this.validateForm(form);
-			if (!valid.valid) {
+			const result = await this.validateForm(form);
```

Comment:
I think we need to remove the attachment validation logic entirely, including the loc string and use an event to externalize the custom validation.

Inside `this.validateForm(form)` we could fire an event:

```ts
validateForm(form) {
  ... current logic to get list of errors ...
  
  // new event:
  this.dispatchEvent(new CustomEvent('form-validating', { detail: { form, errors }, bubbles: true });

  // current logic to return result
  return ...
}
```

This would enable subscribers of the event to push their own custom errors into the errors array to augment the standard validation, removing the need for us to handle tag or attachment internals.

Reply:
Removed everything tag related.

Reply:
Also added another custom event for clearing validation errors so that the subscriber will know when to remove the custom errors.